src/dra/843B0.c

```diff
@@ -417,7 +417,124 @@ INCLUDE_ASM("dra/nonmatchings/843B0", EntityHolyWaterBreakGlass);
 INCLUDE_ASM("dra/nonmatchings/843B0", EntityHolyWaterFlame);
 
 // cross subweapon crash (full effect with all parts)
-INCLUDE_ASM("dra/nonmatchings/843B0", EntitySubwpnCrashCross);
+void EntitySubwpnCrashCross(Entity* self) {
+    Primitive* prim;
+    s16 right;
+    s16 left;
+    s32 var_v0;
+    u16 temp_three;
+    s8 temp_one;
+    u16 three = 3;
+    u32 one = 1;
+
+    func_8010DFF0(1, 1);
+    switch (self->step) {
+    case 0:
+        self->primIndex = AllocPrimitives(PRIM_GT4, 1);
+        if (self->primIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+        self->flags = FLAG_HAS_PRIMS;
+        self->posY.i.hi = 0x78;
+        self->ext.crashcross.unk80 = 1;
+        self->zPriority = 0xC2;
+        func_8011A290(self);
+        LoadImage(&D_800B0788, &D_800B06C8);
```

Comment:
`D_800B06C8` is an array of data. I'd suggest to use it without `&` and change the type as `u16 D_800B06C8[0x30]`. I came up with that type as `D_800B0788` seems to suggest to load 0x30 half words in a row. It most likely contains three palettes.